django/views/decorators/http.py

```diff
@@ -100,10 +118,9 @@ def inner(request, *args, **kwargs):
                 etag=res_etag,
                 last_modified=res_last_modified,
             )
+            return response, res_etag, res_last_modified
 
-            if response is None:
-                response = func(request, *args, **kwargs)
-
+        def _post_process_request(request, response, res_etag, res_last_modified):
```

Comment:
I'm not sure if creating two inline functions is justified for each use of `condition()`. What do you think about moving them to the module level?
```python
def _pre_process_request(request, *args, **kwargs):
    ...

def _post_process_request(request, response, res_etag, res_last_modified):
    ...

def condition(etag_func=None, last_modified_func=None):
    def decorator(func):
        if iscoroutinefunction(func):
            @wraps(func)
            async def inner(request, *args, **kwargs):
                response, res_etag, res_last_modified = _pre_process_request(request, *args, **kwargs)
                if response is None:
                    response = await func(request, *args, **kwargs)
                _post_process_request(request, response, res_etag, res_last_modified)
                return response

        else:

            @wraps(func)
            def inner(request, *args, **kwargs):
                response, res_etag, res_last_modified = _pre_process_request(request, *args, **kwargs)
                if response is None:
                    response = func(request, *args, **kwargs)
                _post_process_request(request, response, res_etag, res_last_modified)
                return response

        return inner
```

Reply:
That's not possible for `_pre_process_request` because it refers to `etag_func` and `last_modified_func`

Reply:
Right :+1: Let's leave them where they are.